classport-analyser/src/test/java/AnalyserTest.java

```diff
@@ -0,0 +1,110 @@
+
+    @Test
+    void testGetSBOM_withAnnotatedClasses() throws IOException {
+        JarFile jar = new JarFile(annotatedJarPath.toFile());
+        HashMap<String, ClassportInfo> actualSbom = Analyser.getSBOM(jar);
+
+        assertTrue(!actualSbom.isEmpty());
+        assertEquals(4, actualSbom.size(), "SBOM should contain 4 annotated classes");
+        assertTrue(actualSbom.containsKey("com.example:test-agent-app:jar:1.0-SNAPSHOT"), "SBOM should contain class com.example:test-agent-app:jar:1.0-SNAPSHOT");
+        assertTrue(actualSbom.containsKey("com.google.errorprone:error_prone_annotations:jar:2.27.0"), "SBOM should contain class com.google.errorprone:error_prone_annotations:jar:2.27.0");
+        assertTrue(actualSbom.containsKey("com.google.code.gson:gson:jar:2.11.0"), "SBOM should contain class com.google.code.gson:gson:jar:2.11.0");
+        assertTrue(actualSbom.containsKey("org.apache.commons:commons-lang3:jar:3.17.0"), "SBOM should contain class org.apache.commons:commons-lang3:jar:3.17.0");   
+    }
+
+    @Test
+    void testGetSBOM_withNonAnnotatedClasses() throws IOException {        
+        ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+        System.setErr(new PrintStream(errContent));
+
+        JarFile jar = new JarFile(notAnnotatedJarPath.toFile());
+        HashMap<String, ClassportInfo> actualSbom = Analyser.getSBOM(jar);
+
+        assertTrue(actualSbom.isEmpty(),"The sbom should be empty if the jar file does not contain annotated classes");
+        String actualMessage = errContent.toString();
+        assertTrue(actualMessage.contains("[Warning]"), "The warning message should match the expected pattern");
+    }
+
+    @Test
+    void testForceClassLoading_withAnnotation() throws IOException, IllegalAccessException, IllegalArgumentException, InvocationTargetException, NoSuchMethodException, SecurityException {
+        Path path = annotatedClassPath;
+        byte[] originalMainClass = Files.readAllBytes(path);
+      
+        String[] classesToLoad = {"com/google/errorprone/annotations/MustBeClosed", "com/google/gson/ExclusionStrategy", "org/apache/commons/lang3/BooleanUtils"};
+        byte[] modifiedClass = ClassLoadingAdder.forceClassLoading(originalMainClass, classesToLoad);
+
+        assertNotNull(modifiedClass, "Modified class byte array should not be null");
+        assertTrue(modifiedClass.length > 0, "Modified class should have content");
+        
+        ClassReader classReader = new ClassReader(modifiedClass);
+        assertTrue(isAdditionalMethodPresent(classReader), "The additional method __classportForceClassLoading should be present");
+        assertTrue(isAdditionalMethodInvoked(classReader), "The additional method __classportForceClassLoading should be invoked");
+        
+    }
+
+    private boolean isAdditionalMethodPresent(ClassReader classReader) {
+        final boolean[] methodPresent = {false}; 
```

Comment:
I used the array because it is in an inner class and it requires a final variable. But If the variable is final I cannot change its value from false to true. Using the array I can declare it final and also modify it. But if you think it is better not to use an array, I can declare a boolean as a class-level field.

Reply:
> I can declare a boolean as a class-level field.

That would be a good coding practice. To declare it as a `private` field.